ocrd/ocrd/cli/resmgr.py

```diff
@@ -0,0 +1,114 @@
+import sys
+from os import getcwd
+from pathlib import Path
+import requests
+
+import click
+
+from ocrd_utils import (
+    initLogging,
+    getLogger,
+    XDG_CACHE_HOME,
+    XDG_CONFIG_HOME,
+    XDG_DATA_HOME
+)
+from ocrd_validators import OcrdZipValidator
+
+from ..resource_manager import OcrdResourceManager
+
+def print_resources(executable, reslist):
+    print('%s' % executable)
+    for resdict in reslist:
+        print('- %s (%s)\n  %s' % (resdict['name'], resdict['url'], resdict['description']))
+    print()
+
+@click.group("resmgr")
+def resmgr_cli():
+    """
+    Managing processor resources
+    """
+    initLogging()
+
+@resmgr_cli.command('list-available')
+@click.option('-e', '--executable', help='Show only resources for executable EXEC', metavar='EXEC')
+def list_available(executable=None):
+    """
+    List available resources
+    """
+    resmgr = OcrdResourceManager()
+    for executable, reslist in resmgr.list_available(executable):
+        print_resources(executable, reslist)
+
+@resmgr_cli.command('list-installed')
+@click.option('-e', '--executable', help='Show only resources for executable EXEC', metavar='EXEC')
+def list_installed(executable=None):
+    """
+    List installed resources
+    """
+    resmgr = OcrdResourceManager()
+    ret = []
+    for executable, reslist in resmgr.list_installed(executable):
+        print_resources(executable, reslist)
+
+@resmgr_cli.command('download')
+@click.option('-n', '--any-url', help='Allow downloading unregistered resources', is_flag=True)
+@click.option('-o', '--overwrite', help='Overwrite existing resources', is_flag=True)
+@click.option('-l', '--location', help='Where to store resources', type=click.Choice(['cache', 'config', 'data', 'cwd']), default='cache', show_default=True)
+@click.argument('executable', required=True)
+@click.argument('url_or_name', required=True)
+def download(any_url, overwrite, location, executable, url_or_name):
+    """
+    Download resource URL_OR_NAME for processor EXECUTABLE.
+
+    URL_OR_NAME can either be the ``name`` or ``url`` of a registered resource.
+
+    If URL_OR_NAME is '*' (asterisk), download all known resources for this processor
+
+    If ``--any-url`` is given, also accepts URL of non-registered resources for ``URL_OR_NAME``.
+    """
+    log = getLogger('ocrd.cli.resmgr')
+    resmgr = OcrdResourceManager()
+    basedir = XDG_CACHE_HOME if location == 'cache' else \
+            XDG_DATA_HOME if location == 'data' else \
+            XDG_CONFIG_HOME if location == 'config' else \
+            getcwd()
+    is_url = url_or_name.startswith('https://') or url_or_name.startswith('http://')
+    find_kwargs = {'executable': executable}
+    if url_or_name != '*':
+        find_kwargs['url' if is_url else 'name'] = url_or_name
+    reslist = resmgr.find_resources(**find_kwargs)
+    if not reslist:
+        log.info("No resources found in registry")
+        if is_url and any_url:
```

Comment:
> For example, I might have mounted my models via NFS or downloaded them via some other means already

You mean, you want to allow absolute file paths in addition to HTTP(S) URL? Sure, that would be doable.

Reply:
Yes, absolute or CWD-relative paths – it does not matter, because you would have to copy (or link) them to a standard location (where `list_all_resources` would expect them) anyway.

Reply:
509200c8